src/SignalR/server/Core/src/HubOptions.cs

```diff
@@ -39,7 +39,17 @@ public class HubOptions
         /// <summary>
         /// Gets or sets the maximum message size of a single incoming hub message. The default is 32KB.
         /// </summary>
-        public long? MaximumReceiveMessageSize { get; set; } = null;
+        public long? MaximumReceiveMessageSize
+        {
+            get => _maximumReceiveMessageSize;
+            set
+            {
+                // null means no-limit and uninitialized, in order to allow the no-limit case
+                // we need to track if the user has set the value
+                _maximumReceiveMessageSizeSet = true;
+                _maximumReceiveMessageSize = value;
```

Comment:
Why is this property the only one that requires this pattern? If someone were to se the HandshakeTimeout to null for example, how would we know whether to disable the handshake timeout or use the globally configured timeout?

Reply:
Because of inconsistent use of null, HandshakeTimeout can't ever be null, we set it to the default and assume it's been set in the HubConnectionContext.